### PR TITLE
docs layout

### PR DESCRIPTION
**Pull request by danethurber**
- Provides basic layout elements(header, aside, main)

### CODE REVIEW

File: docs/gatsby-config.js [hunk 1]
```diff
@@ -4,22 +4,37 @@
...
+  siteMetadata: {
+    title: 'Pluralsight Design System'
+  },
   plugins: [
     {
-      resolve: `gatsby-source-filesystem`,
+      resolve: 'gatsby-source-filesystem',
       options: {
-        name: `pages`,
+        name: 'pages',
```
Comment (danethurber) [inline docs/gatsby-config.js:14]: key if/when we have different layouts per content type

File: docs/src/components/frame/aside.module.css [hunk 2]
```diff
@@ -0,0 +1,58 @@
+@import '@pluralsight/ps-design-system-core';
+
+@custom-media --breakpoint-small (width >= 768px);
```
Comment (danethurber) [inline docs/src/components/frame/aside.module.css:3]: can't figure out a way to use cssvars in media queries 🤷 

also i need to figure out how to make these media definitions globally defined/available

File: docs/src/components/frame/aside.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import useOnEscape from '../../hooks/use-on-escape'
+
+import styles from './aside.module.css'
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  onRequestClose: () => void
+  open: boolean
+}
+
+export const Aside: React.FC<Props> = (props) => {
```
Comment (danethurber) [inline docs/src/components/frame/aside.tsx:16]: i'm doing all named exports for components....thoughts?

File: docs/src/components/layouts/default.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import { MDXProvider } from '@mdx-js/react'
+
+import { Frame } from '../frame'
+import { SideNav } from '../side-nav'
+
+const shortcodes = {}
```
Comment (danethurber) [inline docs/src/components/layouts/default.tsx:7]: this is just a placeholder for if/when we use markdown shortcodes

File: docs/src/hooks/use-match-media.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { useCallback, useEffect, useState } from 'react'
+import { canUseEventListeners } from 'exenv'
+
+import useHasMounted from './use-has-mounted'
+
+export default function useMatchMedia(query: string): boolean {
```
Comment (danethurber) [inline docs/src/hooks/use-match-media.ts:6]: i didn't end up using this hook but i left it in just in case

**Review by danethurber — COMMENTED** [review]
two things i'll fix in follow prs

- general page elements (headings, p, etc) are not theme aware
- eslint in the docs dir is unhappy (typescript integration needed)

File: docs/src/components/frame/aside.tsx (hunk 3 above)
Comment (jaketrent) [reply docs/src/components/frame/aside.tsx:16]: I dig.

File: docs/src/components/frame/aside.module.css (hunk 2 above)
Comment (jaketrent) [reply docs/src/components/frame/aside.module.css:3]: Yeah, I think you found the solution: https://cssnext.github.io/features/#custom-media-queries

Apparently you can't use var() in media queries: https://stackoverflow.com/a/40723269

File: docs/package.json [hunk 6]
```diff
@@ -28,17 +28,27 @@
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
     "gatsby-plugin-express": "^1.1.6",
+    "gatsby-plugin-react-helmet": "^3.3.10",
```
Comment (jaketrent) [inline docs/package.json:31]: magic helmet?! https://www.youtube.com/watch?v=5kH0Bag0akc

File: docs/postcss.config.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+/* eslint-env node */
+/* eslint-disable */
+
+const mediaPlugin = require('postcss-custom-media')
```
Comment (jaketrent) [inline docs/postcss.config.js:4]: looks like a good set to me.

File: docs/src/hooks/use-on-document-click.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react'
+import { canUseEventListeners } from 'exenv'
+
+export default function useOnDocumentClick(
```
Comment (jaketrent) [inline docs/src/hooks/use-on-document-click.ts:4]: I think you could use this instead: https://github.com/pluralsight/design-system/blob/master/packages/util/src/use-close-on-document-events.js

File: docs/src/hooks/use-on-escape.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import { useCallback, useEffect } from 'react'
+import { canUseEventListeners } from 'exenv'
+
+const ESCAPE_KEY = 27
+
+export default function useOnEscape(
```
Comment (jaketrent) [inline docs/src/hooks/use-on-escape.ts:6]: Let's util this instead of making yet another copy.

File: docs/tsconfig.json [hunk 10]
```diff
@@ -1,8 +1,10 @@
 {
   "extends": "../tsconfig.json",
   "compilerOptions": {
+    "jsx": "react",
+    "outDir": "dist",
     "rootDir": "src",
-    "outDir": "dist"
+    "target": "es5"
```
Comment (jaketrent) [inline docs/tsconfig.json:7]: Tell me about how you came to this "target" option.

File: docs/src/hooks/use-has-mounted.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import { useEffect, useRef } from 'react'
```
Comment (jaketrent) [inline docs/src/hooks/use-has-mounted.ts:1]: Util candidate

**Review by jaketrent — APPROVED** [review]
Looks fun!  We get to start fresh.

File: docs/src/hooks/use-on-document-click.ts (hunk 8 above)
Comment (danethurber) [reply docs/src/hooks/use-on-document-click.ts:4]: yep. these are all copied from other parts of the project. i'll follow up with a pr that centralizes them

File: docs/tsconfig.json (hunk 10 above)
Comment (danethurber) [reply docs/tsconfig.json:7]: looking at the gatsby docs

File: docs/typings/pluralsight.d.ts [hunk 12]
```diff
@@ -0,0 +1,8 @@
+declare module '@pluralsight/*'
+
+declare module '@pluralsight/ps-design-system-theme' {
```
[the remark this answers is not in the document]
Comment (danethurber) [reply docs/typings/pluralsight.d.ts:3]: for now i'd just like to fix the ts errors...and then we can publish actual typings with all the packages

**Pull request merged**